[Bug] Version 5.10 is causing "SyntaxError: Unexpected end of JSON input" error

**Describe the bug**
Version 5.10 is causing "SyntaxError: Unexpected end of JSON input" error
![afbeelding](https://user-images.githubusercontent.com/6773985/163355572-aedeabb5-e858-4426-80e9-ac5d38bf508c.png)


Failing build:
![afbeelding](https://user-images.githubusercontent.com/6773985/163355111-0034f03f-c121-4a4e-a6da-2959abcc42e1.png)
Normal working build:
![afbeelding](https://user-images.githubusercontent.com/6773985/163355333-21e2dc15-a630-4a4e-be30-4a829c37f8a1.png)


## Expected Behavior
No errors

## Actual Behavior
I believe this is because since version 5.10 the variable 'Branch from build environment: ' is not being set.

## Possible Fix
Make sure the variable is set again

## Steps to Reproduce
<!--
Just run a build which was working before.
-->

## Context
We always use the newest version fi gitvesion, since the update a few hours ago, our builds stopped working.

## Your Environment
Azure DevOps (private environment)

- Version Used:
Agent name: 'Azure Pipelines 2'
Agent machine name: 'WIN-MGDD5L23JQP'
Current agent version: '2.202.0'
Operating System
Virtual Environment
Virtual Environment Provisioner
Current image version: '20220330.1'
Agent running as: 'VssAdministrator'
Prepare build directory.
Set build variables.
Download all required tasks.
Downloading task: gitversion/setup (0.9.13)
Downloading task: gitversion/execute (0.9.13)
Downloading task: NodeTool (0.200.0)
Downloading task: Npm (1.198.0)
Downloading task: DotNetCoreCLI (2.198.0)
Downloading task: reportgenerator (5.1.4)
Downloading task: PublishCodeCoverageResults (1.200.0)
Downloading task: Gitleaks (2.2.16)
Downloading task: PublishBuildArtifacts (1.200.0)

We are having the same issue on Azure DevOps (using https://github.com/GitTools/actions/issues).

```
2022-04-14T08:54:08.7396902Z ##[section]Starting: Generate version
2022-04-14T08:54:08.7406213Z ==============================================================================
2022-04-14T08:54:08.7406533Z Task         : Execute GitVersion Task
2022-04-14T08:54:08.7406846Z Description  : Easy Semantic Versioning (http://semver.org) for projects using Git
2022-04-14T08:54:08.7407170Z Version      : 0.9.13
2022-04-14T08:54:08.7407355Z Author       : GitTools Contributors
2022-04-14T08:54:08.7407634Z Help         : See the [documentation](https://gitversion.net/docs/) for help
2022-04-14T08:54:08.7408056Z ==============================================================================
2022-04-14T08:54:08.9282914Z Command: dotnet-gitversion /home/vsts/work/1/s /output json /output buildserver /updateprojectfiles
2022-04-14T08:54:08.9306728Z [command]/opt/hostedtoolcache/GitVersion.Tool/5.10.0/x64/dotnet-gitversion /home/vsts/work/1/s /output json /output buildserver /updateprojectfiles
2022-04-14T08:54:09.4952054Z   ERROR [04/14/22 8:54:09:42] An unexpected error occurred:
2022-04-14T08:54:09.4953534Z INFO [04/14/22 8:54:09:15] Working directory: /home/vsts/work/1/s
2022-04-14T08:54:09.4980108Z LibGit2Sharp.LibGit2SharpException: this remote has never connected
2022-04-14T08:54:09.4981179Z INFO [04/14/22 8:54:09:18] Project root is: /home/vsts/work/1/s/
2022-04-14T08:54:09.4981941Z    at LibGit2Sharp.Core.Ensure.HandleError(Int32 result) in /_/LibGit2Sharp/Core/Ensure.cs:line 154
2022-04-14T08:54:09.4982919Z    at LibGit2Sharp.Core.Proxy.git_remote_ls(Repository repository, RemoteHandle remote) in /_/LibGit2Sharp/Core/Proxy.cs:line 2324
2022-04-14T08:54:09.4983870Z    at LibGit2Sharp.Network.ListReferencesInternal(String url, CredentialsHandler credentialsProvider) in /_/LibGit2Sharp/Network.cs:line 130
2022-04-14T08:54:09.4984689Z    at LibGit2Sharp.Network.ListReferences(Remote remote) in /_/LibGit2Sharp/Network.cs:line 55
2022-04-14T08:54:09.4985600Z    at GitVersion.GitRepository.GetPullRequestReference(AuthenticationInfo auth, Remote remote, String headTipSha) in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\Git\GitRepository.cs:line 151
2022-04-14T08:54:09.4986692Z    at GitVersion.GitRepository.<>c__DisplayClass28_0.<CreateBranchForPullRequestBranch>b__0() in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\Git\GitRepository.cs:line 79
2022-04-14T08:54:09.4987946Z    at GitVersion.RepositoryExtensions.RunSafe(Action operation) in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\RepositoryExtensions.cs:line 15
2022-04-14T08:54:09.4988941Z    at GitVersion.GitRepository.CreateBranchForPullRequestBranch(AuthenticationInfo auth) in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\Git\GitRepository.cs:line 69
2022-04-14T08:54:09.4989938Z    at GitVersion.GitPreparer.<>c__DisplayClass18_0.<EnsureHeadIsAttachedToBranch>b__3() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 237
2022-04-14T08:54:09.4990893Z    at GitVersion.Helpers.RetryAction`1.<>c__DisplayClass1_0.<Execute>b__0() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Helpers\RetryAction.cs:line 16
2022-04-14T08:54:09.4991687Z    at Polly.Policy`1.<>c__DisplayClass11_0.<Execute>b__0(Context _, CancellationToken _)
2022-04-14T08:54:09.4992747Z    at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Action`4 onRetry, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider)
2022-04-14T08:54:09.4993861Z    at Polly.Retry.RetryPolicy`1.Implementation(Func`3 action, Context context, CancellationToken cancellationToken)
2022-04-14T08:54:09.4994568Z    at Polly.Policy`1.Execute(Func`3 action, Context context, CancellationToken cancellationToken)
2022-04-14T08:54:09.4995266Z    at Polly.Policy`1.Execute(Func`1 action)
2022-04-14T08:54:09.4995964Z    at GitVersion.Helpers.RetryAction`2.Execute(Func`1 operation) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Helpers\RetryAction.cs:line 35
2022-04-14T08:54:09.4996980Z    at GitVersion.Helpers.RetryAction`1.Execute(Action operation) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Helpers\RetryAction.cs:line 14
2022-04-14T08:54:09.4997890Z    at GitVersion.GitPreparer.EnsureHeadIsAttachedToBranch(String currentBranchName, AuthenticationInfo authentication) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 237
2022-04-14T08:54:09.4998911Z    at GitVersion.GitPreparer.NormalizeGitDirectory(Boolean noFetch, String currentBranchName, Boolean isDynamicRepository) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 183
2022-04-14T08:54:09.4999928Z    at GitVersion.GitPreparer.NormalizeGitDirectory(String targetBranch, Boolean isDynamicRepository) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 134
2022-04-14T08:54:09.5000832Z    at GitVersion.GitPreparer.PrepareInternal(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 69
2022-04-14T08:54:09.5001654Z    at GitVersion.GitPreparer.Prepare() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 48
2022-04-14T08:54:09.5002464Z    at GitVersion.GitVersionCalculateTool.CalculateVersionVariables() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitVersionCalculateTool.cs:line 43
2022-04-14T08:54:09.5004094Z    at GitVersion.GitVersionExecutor.RunGitVersionTool(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.App\GitVersionExecutor.cs:line 66
2022-04-14T08:54:09.5017364Z INFO [04/14/22 8:54:09:18] DotGit directory is: /home/vsts/work/1/s/.git
2022-04-14T08:54:09.5017976Z INFO [04/14/22 8:54:09:18] Branch from build environment: 
2022-04-14T08:54:09.5019120Z INFO [04/14/22 8:54:09:18] Begin: Normalizing git directory for branch ''
2022-04-14T08:54:09.5020051Z   INFO [04/14/22 8:54:09:21] One remote found (origin -> REDACTED).
2022-04-14T08:54:09.5020879Z   INFO [04/14/22 8:54:09:22] Skipping fetching, if GitVersion does not calculate your version as expected you might need to allow fetching or use dynamic repositories
2022-04-14T08:54:09.5022168Z   INFO [04/14/22 8:54:09:23] Creating local branch from remote tracking 'refs/remotes/origin/REDACTED'.
2022-04-14T08:54:09.5047288Z   INFO [04/14/22 8:54:09:28] Creating local branch from remote tracking 'refs/remotes/origin/main'.
2022-04-14T08:54:09.5048016Z   INFO [04/14/22 8:54:09:29] HEAD is detached and points at commit 'e07dc44f60a99c545e6d34cf9e3dfc56289753bd'.
2022-04-14T08:54:09.5048435Z   INFO [04/14/22 8:54:09:29] Local Refs:
2022-04-14T08:54:09.5049116Z refs/heads/ (de822a25f32e270730492a0aa2b14e36a436d2c0)
2022-04-14T08:54:09.5049951Z refs/heads/ (4af01019cd4a86250021bd28a9950d86737d0279)
2022-04-14T08:54:09.5050802Z refs/heads/ (772a918fade02c7e45d7b4077639b605dc1627aa)
2022-04-14T08:54:09.5051616Z refs/heads/ (103fba4fa8192b9a773930037eb7dbe69fa85c06)
2022-04-14T08:54:09.5052400Z refs/heads/ (5bbf919fdb6e3a0f536644ca1124aa38c1bb6a2a)
2022-04-14T08:54:09.5053201Z refs/heads/ (4e81a6c88fcd52cf7073535b2c407caf9c75e404)
2022-04-14T08:54:09.5054048Z refs/heads/ (d064f3fe3c42afe3c94e6e5054e8e7d6e4490e7a)
2022-04-14T08:54:09.5054899Z refs/heads/ (d7c24bf33ab3fea05fc2bd93f97fd7ff4f440768)
2022-04-14T08:54:09.5055704Z refs/heads/ (30e7771a718bea48d17eb07ea0460c24c4317396)
2022-04-14T08:54:09.5056775Z refs/heads/ (e233c898a9a69f40538505b026dd3fc6f715bb6e)
2022-04-14T08:54:09.5057587Z refs/heads/ (9db3aca5bb2500f9b8c89930c44c362308022386)
2022-04-14T08:54:09.5058389Z refs/heads/ (ddd782d9192b52ee56bb3eaafe69dcfe093fa354)
2022-04-14T08:54:09.5059176Z refs/heads/ (339c3c72e25af822488af4fc6429351841fd86d8)
2022-04-14T08:54:09.5059971Z refs/heads/ (70f628609f325d29342365a9139cdba80fc30f79)
2022-04-14T08:54:09.5060777Z refs/heads/ (1064d0aedd18c9cd71fdebe08723b84cd8ff4502)
2022-04-14T08:54:09.5061661Z refs/heads/ (029f7c6903924ca254a320b10ab09fbdc4942f99)
2022-04-14T08:54:09.5062469Z refs/heads/ (57f3a5c46dcaae462a35333516f7769bb5a7ffe9)
2022-04-14T08:54:09.5063226Z refs/heads/ (ee65c46734d376115f61481e5b16e12cdbc0ecbd)
2022-04-14T08:54:09.5063947Z refs/heads/ (3b59e29a1c2641cee8a1ee80859630e0163de626)
2022-04-14T08:54:09.5064577Z refs/heads/ (60170ad8dd7e0eb18495fe572e1b275c632502e8)
2022-04-14T08:54:09.5065209Z refs/heads/ (4382b5668cb55cd74eab1ae8f7a880c49c4fb1bc)
2022-04-14T08:54:09.5065838Z refs/heads/ (d333724501a3fd42e4f8db696f62a851947550ba)
2022-04-14T08:54:09.5066494Z refs/heads/ (e17828da50e9bc3fb6ef8a7301665000a0eb7ba6)
2022-04-14T08:54:09.5066909Z refs/heads/main (f75e9fabb91aedbaa1a42467d3467d8a00e44d92)
2022-04-14T08:54:09.5067647Z refs/remotes/origin/ (de822a25f32e270730492a0aa2b14e36a436d2c0)
2022-04-14T08:54:09.5068515Z refs/remotes/origin/ (4af01019cd4a86250021bd28a9950d86737d0279)
2022-04-14T08:54:09.5069505Z refs/remotes/origin/ (772a918fade02c7e45d7b4077639b605dc1627aa)
2022-04-14T08:54:09.5070351Z refs/remotes/origin/ (103fba4fa8192b9a773930037eb7dbe69fa85c06)
2022-04-14T08:54:09.5071163Z refs/remotes/origin/ (5bbf919fdb6e3a0f536644ca1124aa38c1bb6a2a)
2022-04-14T08:54:09.5072001Z refs/remotes/origin/ (4e81a6c88fcd52cf7073535b2c407caf9c75e404)
2022-04-14T08:54:09.5072877Z refs/remotes/origin/ (d064f3fe3c42afe3c94e6e5054e8e7d6e4490e7a)
2022-04-14T08:54:09.5073768Z refs/remotes/origin/ (d7c24bf33ab3fea05fc2bd93f97fd7ff4f440768)
2022-04-14T08:54:09.5074618Z refs/remotes/origin/ (30e7771a718bea48d17eb07ea0460c24c4317396)
2022-04-14T08:54:09.5075470Z refs/remotes/origin/ (e233c898a9a69f40538505b026dd3fc6f715bb6e)
2022-04-14T08:54:09.5076301Z refs/remotes/origin/ (9db3aca5bb2500f9b8c89930c44c362308022386)
2022-04-14T08:54:09.5077147Z refs/remotes/origin/ (ddd782d9192b52ee56bb3eaafe69dcfe093fa354)
2022-04-14T08:54:09.5078014Z refs/remotes/origin/ (339c3c72e25af822488af4fc6429351841fd86d8)
2022-04-14T08:54:09.5078934Z refs/remotes/origin/ (70f628609f325d29342365a9139cdba80fc30f79)
2022-04-14T08:54:09.5079778Z refs/remotes/origin/ (1064d0aedd18c9cd71fdebe08723b84cd8ff4502)
2022-04-14T08:54:09.5080682Z refs/remotes/origin/ (029f7c6903924ca254a320b10ab09fbdc4942f99)
2022-04-14T08:54:09.5081530Z refs/remotes/origin/ (57f3a5c46dcaae462a35333516f7769bb5a7ffe9)
2022-04-14T08:54:09.5082336Z refs/remotes/origin/ (ee65c46734d376115f61481e5b16e12cdbc0ecbd)
2022-04-14T08:54:09.5083075Z refs/remotes/origin/ (3b59e29a1c2641cee8a1ee80859630e0163de626)
2022-04-14T08:54:09.5083742Z refs/remotes/origin/ (60170ad8dd7e0eb18495fe572e1b275c632502e8)
2022-04-14T08:54:09.5084406Z refs/remotes/origin/ (4382b5668cb55cd74eab1ae8f7a880c49c4fb1bc)
2022-04-14T08:54:09.5085120Z refs/remotes/origin/ (d333724501a3fd42e4f8db696f62a851947550ba)
2022-04-14T08:54:09.5085845Z refs/remotes/origin/ (e17828da50e9bc3fb6ef8a7301665000a0eb7ba6)
2022-04-14T08:54:09.5086287Z refs/remotes/origin/main (f75e9fabb91aedbaa1a42467d3467d8a00e44d92)
2022-04-14T08:54:09.5086701Z refs/remotes/pull/2370/merge (e07dc44f60a99c545e6d34cf9e3dfc56289753bd)
2022-04-14T08:54:09.5087461Z   INFO [04/14/22 8:54:09:29] No local branch pointing at the commit 'e07dc44f60a99c545e6d34cf9e3dfc56289753bd'. Fake branch needs to be created.
2022-04-14T08:54:09.5088016Z   INFO [04/14/22 8:54:09:29] Fetching remote refs to see if there is a pull request ref
2022-04-14T08:54:09.5088641Z   INFO [04/14/22 8:54:09:39] End: Normalizing git directory for branch '' (Took: 209.79ms)
2022-04-14T08:54:09.5089111Z   INFO [04/14/22 8:54:09:42] Attempting to show the current git graph (please include in issue): 
2022-04-14T08:54:09.5089530Z   INFO [04/14/22 8:54:09:42] Showing max of 100 commits
2022-04-14T08:54:09.5089984Z   INFO [04/14/22 8:54:09:47] *   e07dc44 14 minutes ago  (HEAD, pull/2370/merge)
2022-04-14T08:54:09.5090290Z |\  
2022-04-14T08:54:09.5090825Z | *   e17828d 14 minutes ago  ()
2022-04-14T08:54:09.5091169Z | |\  
2022-04-14T08:54:09.5091376Z | | * db6df6d 42 minutes ago 
2022-04-14T08:54:09.5091619Z | * | ec9544b 14 minutes ago 
2022-04-14T08:54:09.5091838Z |/ /  
2022-04-14T08:54:09.5092076Z * / f75e9fa 16 minutes ago  (origin/main, main)
2022-04-14T08:54:09.5092312Z |/  
2022-04-14T08:54:09.5093061Z | * 1064d0a 3 hours ago  ()
2022-04-14T08:54:09.5093574Z | * e4ef0e4 7 hours ago 
2022-04-14T08:54:09.5093785Z |/  
2022-04-14T08:54:09.5094581Z | * 029f7c6 6 hours ago  ()
2022-04-14T08:54:09.5095151Z | * 9705fc6 7 hours ago 
2022-04-14T08:54:09.5095353Z |/  
2022-04-14T08:54:09.5096012Z | * ee65c46 6 hours ago  ()
2022-04-14T08:54:09.5099139Z | * c2e1f95 7 hours ago 
2022-04-14T08:54:09.5099486Z |/  
2022-04-14T08:54:09.5100328Z | * 70f6286 6 hours ago  ()
2022-04-14T08:54:09.5100802Z | * eb56393 7 hours ago 
2022-04-14T08:54:09.5101021Z |/  
2022-04-14T08:54:09.5101716Z | * ddd782d 6 hours ago  ()
2022-04-14T08:54:09.5102359Z | * 5877c39 7 hours ago 
2022-04-14T08:54:09.5102567Z |/  
2022-04-14T08:54:09.5103297Z | * 30e7771 6 hours ago  ()
2022-04-14T08:54:09.5103788Z | * bf2766b 7 hours ago 
2022-04-14T08:54:09.5104000Z |/  
2022-04-14T08:54:09.5104201Z * 90a373e 19 hours ago 
2022-04-14T08:54:09.5104432Z * 9598f1d 21 hours ago 
2022-04-14T08:54:09.5104678Z * 1d117a0 24 hours ago 
2022-04-14T08:54:09.5104910Z * f617959 24 hours ago 
2022-04-14T08:54:09.5105156Z * ee80c4d 24 hours ago 
2022-04-14T08:54:09.5105386Z * 3aa63db 24 hours ago 
2022-04-14T08:54:09.5105616Z * 9826c66 24 hours ago 
2022-04-14T08:54:09.5105864Z * 165b485 26 hours ago 
2022-04-14T08:54:09.5106584Z | * 9db3aca 25 hours ago  ()
2022-04-14T08:54:09.5107088Z | * 2045b8a 31 hours ago 
2022-04-14T08:54:09.5107288Z |/  
2022-04-14T08:54:09.5107981Z | * 339c3c7 31 hours ago  ()
2022-04-14T08:54:09.5108455Z | * 702a590 31 hours ago 
2022-04-14T08:54:09.5108653Z |/  
2022-04-14T08:54:09.5109374Z | * d7c24bf 31 hours ago  ()
2022-04-14T08:54:09.5109869Z | * 097c3a9 31 hours ago 
2022-04-14T08:54:09.5110083Z |/  
2022-04-14T08:54:09.5110278Z * baef284 2 days ago 
2022-04-14T08:54:09.5110517Z * 1b67410 2 days ago 
2022-04-14T08:54:09.5110741Z * 173cdae 2 days ago 
2022-04-14T08:54:09.5111221Z * 60170ad 2 days ago  ()
2022-04-14T08:54:09.5111560Z * a4d25a8 2 days ago 
2022-04-14T08:54:09.5111784Z * ee143eb 2 days ago 
2022-04-14T08:54:09.5112513Z | * 5bbf919 2 days ago  ()
2022-04-14T08:54:09.5113077Z | * 0f85b30 2 days ago 
2022-04-14T08:54:09.5113319Z | * 844ef3a 2 days ago 
2022-04-14T08:54:09.5113545Z | * abd6960 2 days ago 
2022-04-14T08:54:09.5113737Z |/  
2022-04-14T08:54:09.5114525Z | * 772a918 2 days ago  ()
2022-04-14T08:54:09.5115047Z | * c181516 2 days ago 
2022-04-14T08:54:09.5115288Z | * f1f12af 2 days ago 
2022-04-14T08:54:09.5115481Z |/  
2022-04-14T08:54:09.5116144Z | * 4e81a6c 2 days ago  ()
2022-04-14T08:54:09.5116628Z | * 81b13d7 2 days ago 
2022-04-14T08:54:09.5116853Z | * d89faa9 2 days ago 
2022-04-14T08:54:09.5117093Z | * e04f090 2 days ago 
2022-04-14T08:54:09.5117321Z | * 3423107 2 days ago 
2022-04-14T08:54:09.5117531Z |/  
2022-04-14T08:54:09.5118171Z | * 103fba4 2 days ago  ()
2022-04-14T08:54:09.5118615Z | * ce91a41 2 days ago 
2022-04-14T08:54:09.5118823Z |/  
2022-04-14T08:54:09.5119567Z | * 4af0101 2 days ago  ()
2022-04-14T08:54:09.5120093Z | * 2fdeef1 2 days ago 
2022-04-14T08:54:09.5120286Z |/  
2022-04-14T08:54:09.5121002Z | * de822a2 2 days ago  ()
2022-04-14T08:54:09.5123665Z | * eb18082 2 days ago 
2022-04-14T08:54:09.5123905Z |/  
2022-04-14T08:54:09.5124120Z * 6586871 3 days ago 
2022-04-14T08:54:09.5124345Z * b9c6931 3 days ago 
2022-04-14T08:54:09.5124577Z * b8c353e 3 days ago 
2022-04-14T08:54:09.5124818Z * 6432d78 3 days ago 
2022-04-14T08:54:09.5125041Z * cf79a59 3 days ago 
2022-04-14T08:54:09.5125278Z * a2ad971 3 days ago 
2022-04-14T08:54:09.5126055Z | * 57f3a5c 3 days ago  ()
2022-04-14T08:54:09.5126516Z | * 55c1de9 5 days ago 
2022-04-14T08:54:09.5126711Z |/  
2022-04-14T08:54:09.5127397Z | * e233c89 5 days ago  ()
2022-04-14T08:54:09.5127893Z | * 3f6f3b3 5 days ago 
2022-04-14T08:54:09.5128088Z |/  
2022-04-14T08:54:09.5128297Z * 7650e81 6 days ago 
2022-04-14T08:54:09.5128526Z * 08fe9c1 6 days ago 
2022-04-14T08:54:09.5128751Z * f59e5ff 6 days ago 
2022-04-14T08:54:09.5128988Z * 01b5126 6 days ago 
2022-04-14T08:54:09.5129211Z * 6289552 6 days ago 
2022-04-14T08:54:09.5129453Z * a0e9a90 6 days ago 
2022-04-14T08:54:09.5129678Z * 834e427 7 days ago 
2022-04-14T08:54:09.5129902Z * 51d5f88 7 days ago 
2022-04-14T08:54:09.5130142Z * 2f5f906 7 days ago 
2022-04-14T08:54:09.5130364Z * 38a8abf 7 days ago 
2022-04-14T08:54:09.5130586Z * b5360c2 7 days ago 
2022-04-14T08:54:09.5130823Z * d5392ac 7 days ago 
2022-04-14T08:54:09.5131047Z * f09bdf9 7 days ago 
2022-04-14T08:54:09.5131282Z * 5d111d5 8 days ago 
2022-04-14T08:54:09.5131507Z * e099f73 8 days ago 
2022-04-14T08:54:09.5131730Z * 83bd10e 8 days ago 
2022-04-14T08:54:09.5131967Z * b66ace4 8 days ago 
2022-04-14T08:54:09.5132189Z * 4d1c8e7 8 days ago 
2022-04-14T08:54:09.5132429Z * d7c377d 8 days ago 
2022-04-14T08:54:09.5132652Z * 7c621cf 9 days ago 
2022-04-14T08:54:09.5132876Z * 02121ed 9 days ago 
2022-04-14T08:54:09.5133115Z * 782fc41 9 days ago 
2022-04-14T08:54:09.5133340Z * 58560de 9 days ago 
2022-04-14T08:54:09.5133575Z * 9844882 9 days ago 
2022-04-14T08:54:09.5133800Z * 29832d7 9 days ago 
2022-04-14T08:54:09.5134139Z * 3d8424a 9 days ago 
2022-04-14T08:54:09.5134379Z * bf89484 9 days ago 
2022-04-14T08:54:09.5134604Z * 5d9b941 10 days ago 
2022-04-14T08:54:09.5134988Z * 14bf27e 10 days ago 
2022-04-14T08:54:09.5135216Z * bd205a8 10 days ago 
2022-04-14T08:54:09.5135996Z * 7577f18 13 days ago 
2022-04-14T08:54:09.5136230Z * f7c073e 13 days ago 
2022-04-14T08:54:09.5136575Z * 53d20fe 13 days ago 
2022-04-14T08:54:09.5137037Z * 0dcaf92 13 days ago 
2022-04-14T08:54:09.5137267Z * 75f75b6 13 days ago 
2022-04-14T08:54:09.5137389Z 
2022-04-14T08:54:09.5138105Z INFO [04/14/22 8:54:09:09] Applicable build agent found: 'AzurePipelines'.
2022-04-14T08:54:09.5138512Z INFO [04/14/22 8:54:09:15] Working directory: /home/vsts/work/1/s
2022-04-14T08:54:09.5138902Z INFO [04/14/22 8:54:09:18] Project root is: /home/vsts/work/1/s/
2022-04-14T08:54:09.5139289Z INFO [04/14/22 8:54:09:18] DotGit directory is: /home/vsts/work/1/s/.git
2022-04-14T08:54:09.5139677Z INFO [04/14/22 8:54:09:18] Branch from build environment: 
2022-04-14T08:54:09.5140225Z INFO [04/14/22 8:54:09:18] Begin: Normalizing git directory for branch ''
2022-04-14T08:54:09.5140992Z   INFO [04/14/22 8:54:09:21] One remote found (origin -> 'REDACTED').
2022-04-14T08:54:09.5141661Z   INFO [04/14/22 8:54:09:22] Skipping fetching, if GitVersion does not calculate your version as expected you might need to allow fetching or use dynamic repositories
2022-04-14T08:54:09.5142642Z   INFO [04/14/22 8:54:09:23] Creating local branch from remote tracking 'refs/remotes/origin/REDACTED'
2022-04-14T08:54:09.5164786Z   INFO [04/14/22 8:54:09:29] HEAD is detached and points at commit 'e07dc44f60a99c545e6d34cf9e3dfc56289753bd'.
2022-04-14T08:54:09.5165288Z   INFO [04/14/22 8:54:09:29] Local Refs:
2022-04-14T08:54:09.5165959Z refs/heads/(de822a25f32e270730492a0aa2b14e36a436d2c0)
2022-04-14T08:54:09.5166803Z refs/heads/ (4af01019cd4a86250021bd28a9950d86737d0279)
2022-04-14T08:54:09.5167648Z refs/heads/(772a918fade02c7e45d7b4077639b605dc1627aa)
2022-04-14T08:54:09.5168454Z refs/heads/ (103fba4fa8192b9a773930037eb7dbe69fa85c06)
2022-04-14T08:54:09.5169228Z refs/heads/ (5bbf919fdb6e3a0f536644ca1124aa38c1bb6a2a)
2022-04-14T08:54:09.5170013Z refs/heads/ (4e81a6c88fcd52cf7073535b2c407caf9c75e404)
2022-04-14T08:54:09.5170875Z refs/heads/ (d064f3fe3c42afe3c94e6e5054e8e7d6e4490e7a)
2022-04-14T08:54:09.5171714Z refs/heads/ (d7c24bf33ab3fea05fc2bd93f97fd7ff4f440768)
2022-04-14T08:54:09.5172528Z refs/heads/ (30e7771a718bea48d17eb07ea0460c24c4317396)
2022-04-14T08:54:09.5173322Z refs/heads/ (e233c898a9a69f40538505b026dd3fc6f715bb6e)
2022-04-14T08:54:09.5174125Z refs/heads/ (9db3aca5bb2500f9b8c89930c44c362308022386)
2022-04-14T08:54:09.5174909Z refs/heads/ (ddd782d9192b52ee56bb3eaafe69dcfe093fa354)
2022-04-14T08:54:09.5175701Z refs/heads/ (339c3c72e25af822488af4fc6429351841fd86d8)
2022-04-14T08:54:09.5176622Z refs/heads/ (70f628609f325d29342365a9139cdba80fc30f79)
2022-04-14T08:54:09.5177539Z refs/heads/ (1064d0aedd18c9cd71fdebe08723b84cd8ff4502)
2022-04-14T08:54:09.5178392Z refs/heads/ (029f7c6903924ca254a320b10ab09fbdc4942f99)
2022-04-14T08:54:09.5179208Z refs/heads/ (57f3a5c46dcaae462a35333516f7769bb5a7ffe9)
2022-04-14T08:54:09.5179957Z refs/heads/ (ee65c46734d376115f61481e5b16e12cdbc0ecbd)
2022-04-14T08:54:09.5180658Z refs/heads/ (3b59e29a1c2641cee8a1ee80859630e0163de626)
2022-04-14T08:54:09.5181297Z refs/heads/ (60170ad8dd7e0eb18495fe572e1b275c632502e8)
2022-04-14T08:54:09.5181910Z refs/heads/ (4382b5668cb55cd74eab1ae8f7a880c49c4fb1bc)
2022-04-14T08:54:09.5182555Z refs/heads/ (d333724501a3fd42e4f8db696f62a851947550ba)
2022-04-14T08:54:09.5183192Z refs/heads/ (e17828da50e9bc3fb6ef8a7301665000a0eb7ba6)
2022-04-14T08:54:09.5183618Z refs/heads/main (f75e9fabb91aedbaa1a42467d3467d8a00e44d92)
2022-04-14T08:54:09.5184338Z refs/remotes/origin/ (de822a25f32e270730492a0aa2b14e36a436d2c0)
2022-04-14T08:54:09.5185218Z refs/remotes/origin/ (4af01019cd4a86250021bd28a9950d86737d0279)
2022-04-14T08:54:09.5186099Z refs/remotes/origin/ (772a918fade02c7e45d7b4077639b605dc1627aa)
2022-04-14T08:54:09.5186924Z refs/remotes/origin/ (103fba4fa8192b9a773930037eb7dbe69fa85c06)
2022-04-14T08:54:09.5187833Z refs/remotes/origin/ (5bbf919fdb6e3a0f536644ca1124aa38c1bb6a2a)
2022-04-14T08:54:09.5188665Z refs/remotes/origin/ (4e81a6c88fcd52cf7073535b2c407caf9c75e404)
2022-04-14T08:54:09.5189566Z refs/remotes/origin/ (d064f3fe3c42afe3c94e6e5054e8e7d6e4490e7a)
2022-04-14T08:54:09.5190440Z refs/remotes/origin/ (d7c24bf33ab3fea05fc2bd93f97fd7ff4f440768)
2022-04-14T08:54:09.5191292Z refs/remotes/origin/ (30e7771a718bea48d17eb07ea0460c24c4317396)
2022-04-14T08:54:09.5192117Z refs/remotes/origin/ (e233c898a9a69f40538505b026dd3fc6f715bb6e)
2022-04-14T08:54:09.5192957Z refs/remotes/origin/ (9db3aca5bb2500f9b8c89930c44c362308022386)
2022-04-14T08:54:09.5193780Z refs/remotes/origin/ (ddd782d9192b52ee56bb3eaafe69dcfe093fa354)
2022-04-14T08:54:09.5194607Z refs/remotes/origin/ (339c3c72e25af822488af4fc6429351841fd86d8)
2022-04-14T08:54:09.5195414Z refs/remotes/origin/ (70f628609f325d29342365a9139cdba80fc30f79)
2022-04-14T08:54:09.5196266Z refs/remotes/origin/ (1064d0aedd18c9cd71fdebe08723b84cd8ff4502)
2022-04-14T08:54:09.5197154Z refs/remotes/origin/ (029f7c6903924ca254a320b10ab09fbdc4942f99)
2022-04-14T08:54:09.5198008Z refs/remotes/origin/ (57f3a5c46dcaae462a35333516f7769bb5a7ffe9)
2022-04-14T08:54:09.5198792Z refs/remotes/origin/ (ee65c46734d376115f61481e5b16e12cdbc0ecbd)
2022-04-14T08:54:09.5199625Z refs/remotes/origin/ (3b59e29a1c2641cee8a1ee80859630e0163de626)
2022-04-14T08:54:09.5200287Z refs/remotes/origin/ (60170ad8dd7e0eb18495fe572e1b275c632502e8)
2022-04-14T08:54:09.5200934Z refs/remotes/origin/ (4382b5668cb55cd74eab1ae8f7a880c49c4fb1bc)
2022-04-14T08:54:09.5201610Z refs/remotes/origin/ (d333724501a3fd42e4f8db696f62a851947550ba)
2022-04-14T08:54:09.5202277Z refs/remotes/origin/ (e17828da50e9bc3fb6ef8a7301665000a0eb7ba6)
2022-04-14T08:54:09.5202732Z refs/remotes/origin/main (f75e9fabb91aedbaa1a42467d3467d8a00e44d92)
2022-04-14T08:54:09.5203134Z refs/remotes/pull/2370/merge (e07dc44f60a99c545e6d34cf9e3dfc56289753bd)
2022-04-14T08:54:09.5203904Z   INFO [04/14/22 8:54:09:29] No local branch pointing at the commit 'e07dc44f60a99c545e6d34cf9e3dfc56289753bd'. Fake branch needs to be created.
2022-04-14T08:54:09.5204445Z   INFO [04/14/22 8:54:09:29] Fetching remote refs to see if there is a pull request ref
2022-04-14T08:54:09.5205082Z   INFO [04/14/22 8:54:09:39] End: Normalizing git directory for branch '' (Took: 209.79ms)
2022-04-14T08:54:09.5205489Z   ERROR [04/14/22 8:54:09:42] An unexpected error occurred:
2022-04-14T08:54:09.5205859Z LibGit2Sharp.LibGit2SharpException: this remote has never connected
2022-04-14T08:54:09.5206315Z    at LibGit2Sharp.Core.Ensure.HandleError(Int32 result) in /_/LibGit2Sharp/Core/Ensure.cs:line 154
2022-04-14T08:54:09.5206863Z    at LibGit2Sharp.Core.Proxy.git_remote_ls(Repository repository, RemoteHandle remote) in /_/LibGit2Sharp/Core/Proxy.cs:line 2324
2022-04-14T08:54:09.5207494Z    at LibGit2Sharp.Network.ListReferencesInternal(String url, CredentialsHandler credentialsProvider) in /_/LibGit2Sharp/Network.cs:line 130
2022-04-14T08:54:09.5208119Z    at LibGit2Sharp.Network.ListReferences(Remote remote) in /_/LibGit2Sharp/Network.cs:line 55
2022-04-14T08:54:09.5208818Z    at GitVersion.GitRepository.GetPullRequestReference(AuthenticationInfo auth, Remote remote, String headTipSha) in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\Git\GitRepository.cs:line 151
2022-04-14T08:54:09.5209669Z    at GitVersion.GitRepository.<>c__DisplayClass28_0.<CreateBranchForPullRequestBranch>b__0() in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\Git\GitRepository.cs:line 79
2022-04-14T08:54:09.5210455Z    at GitVersion.RepositoryExtensions.RunSafe(Action operation) in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\RepositoryExtensions.cs:line 15
2022-04-14T08:54:09.5211216Z    at GitVersion.GitRepository.CreateBranchForPullRequestBranch(AuthenticationInfo auth) in D:\a\GitVersion\GitVersion\src\GitVersion.LibGit2Sharp\Git\GitRepository.cs:line 69
2022-04-14T08:54:09.5212018Z    at GitVersion.GitPreparer.<>c__DisplayClass18_0.<EnsureHeadIsAttachedToBranch>b__3() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 237
2022-04-14T08:54:09.5212764Z    at GitVersion.Helpers.RetryAction`1.<>c__DisplayClass1_0.<Execute>b__0() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Helpers\RetryAction.cs:line 16
2022-04-14T08:54:09.5213365Z    at Polly.Policy`1.<>c__DisplayClass11_0.<Execute>b__0(Context _, CancellationToken _)
2022-04-14T08:54:09.5214225Z    at Polly.Retry.RetryEngine.Implementation[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Action`4 onRetry, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider)
2022-04-14T08:54:09.5215151Z    at Polly.Retry.RetryPolicy`1.Implementation(Func`3 action, Context context, CancellationToken cancellationToken)
2022-04-14T08:54:09.5215663Z    at Polly.Policy`1.Execute(Func`3 action, Context context, CancellationToken cancellationToken)
2022-04-14T08:54:09.5216068Z    at Polly.Policy`1.Execute(Func`1 action)
2022-04-14T08:54:09.5216809Z    at GitVersion.Helpers.RetryAction`2.Execute(Func`1 operation) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Helpers\RetryAction.cs:line 35
2022-04-14T08:54:09.5217524Z    at GitVersion.Helpers.RetryAction`1.Execute(Action operation) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Helpers\RetryAction.cs:line 14
2022-04-14T08:54:09.5218309Z    at GitVersion.GitPreparer.EnsureHeadIsAttachedToBranch(String currentBranchName, AuthenticationInfo authentication) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 237
2022-04-14T08:54:09.5219196Z    at GitVersion.GitPreparer.NormalizeGitDirectory(Boolean noFetch, String currentBranchName, Boolean isDynamicRepository) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 183
2022-04-14T08:54:09.5220001Z    at GitVersion.GitPreparer.NormalizeGitDirectory(String targetBranch, Boolean isDynamicRepository) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 134
2022-04-14T08:54:09.5220790Z    at GitVersion.GitPreparer.PrepareInternal(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 69
2022-04-14T08:54:09.5221452Z    at GitVersion.GitPreparer.Prepare() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitPreparer.cs:line 48
2022-04-14T08:54:09.5222134Z    at GitVersion.GitVersionCalculateTool.CalculateVersionVariables() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitVersionCalculateTool.cs:line 43
2022-04-14T08:54:09.5222892Z    at GitVersion.GitVersionExecutor.RunGitVersionTool(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.App\GitVersionExecutor.cs:line 66
2022-04-14T08:54:09.5223534Z   INFO [04/14/22 8:54:09:42] Attempting to show the current git graph (please include in issue): 
2022-04-14T08:54:09.5224015Z   INFO [04/14/22 8:54:09:42] Showing max of 100 commits
2022-04-14T08:54:09.5224410Z   INFO [04/14/22 8:54:09:47] *   e07dc44 14 minutes ago  (HEAD, pull/2370/merge)
2022-04-14T08:54:09.5224700Z |\  
2022-04-14T08:54:09.5225275Z | *   e17828d 14 minutes ago  (REDACTED)
2022-04-14T08:54:09.5225617Z | |\  
2022-04-14T08:54:09.5225827Z | | * db6df6d 42 minutes ago 
2022-04-14T08:54:09.5226083Z | * | ec9544b 14 minutes ago 
2022-04-14T08:54:09.5226287Z |/ /  
2022-04-14T08:54:09.5226538Z * / f75e9fa 16 minutes ago  (origin/main, main)
2022-04-14T08:54:09.5226774Z |/  
2022-04-14T08:54:09.5227503Z | * 1064d0a 3 hours ago  (REDACTED)
2022-04-14T08:54:09.5228023Z | * e4ef0e4 7 hours ago 
2022-04-14T08:54:09.5228219Z |/  
2022-04-14T08:54:09.5229027Z | * 029f7c6 6 hours ago  (REDACTED)
2022-04-14T08:54:09.5229581Z | * 9705fc6 7 hours ago 
2022-04-14T08:54:09.5229795Z |/  
2022-04-14T08:54:09.5230442Z | * ee65c46 6 hours ago  (REDACTED)
2022-04-14T08:54:09.5230906Z | * c2e1f95 7 hours ago 
2022-04-14T08:54:09.5231106Z |/  
2022-04-14T08:54:09.5231776Z | * 70f6286 6 hours ago  (REDACTED)
2022-04-14T08:54:09.5232264Z | * eb56393 7 hours ago 
2022-04-14T08:54:09.5232463Z |/  
2022-04-14T08:54:09.5233169Z | * ddd782d 6 hours ago  (REDACTED)
2022-04-14T08:54:09.5233653Z | * 5877c39 7 hours ago 
2022-04-14T08:54:09.5233866Z |/  
2022-04-14T08:54:09.5234566Z | * 30e7771 6 hours ago  (REDACTED)
2022-04-14T08:54:09.5235171Z | * bf2766b 7 hours ago 
2022-04-14T08:54:09.5235370Z |/  
2022-04-14T08:54:09.5235586Z * 90a373e 19 hours ago 
2022-04-14T08:54:09.5235821Z * 9598f1d 21 hours ago 
2022-04-14T08:54:09.5236052Z * 1d117a0 24 hours ago 
2022-04-14T08:54:09.5236299Z * f617959 24 hours ago 
2022-04-14T08:54:09.5236531Z * ee80c4d 24 hours ago 
2022-04-14T08:54:09.5236775Z * 3aa63db 24 hours ago 
2022-04-14T08:54:09.5237007Z * 9826c66 24 hours ago 
2022-04-14T08:54:09.5237237Z * 165b485 26 hours ago 
2022-04-14T08:54:09.5237982Z | * 9db3aca 25 hours ago  (REDACTED)
2022-04-14T08:54:09.5238470Z | * 2045b8a 31 hours ago 
2022-04-14T08:54:09.5238685Z |/  
2022-04-14T08:54:09.5239362Z | * 339c3c7 31 hours ago  (REDACTED)
2022-04-14T08:54:09.5239853Z | * 702a590 31 hours ago 
2022-04-14T08:54:09.5240054Z |/  
2022-04-14T08:54:09.5240765Z | * d7c24bf 31 hours ago  (REDACTED)
2022-04-14T08:54:09.5241274Z | * 097c3a9 31 hours ago 
2022-04-14T08:54:09.5241474Z |/  
2022-04-14T08:54:09.5241686Z * baef284 2 days ago 
2022-04-14T08:54:09.5241911Z * 1b67410 2 days ago 
2022-04-14T08:54:09.5242134Z * 173cdae 2 days ago 
2022-04-14T08:54:09.5242670Z * 60170ad 2 days ago  (REDACTED)
2022-04-14T08:54:09.5242991Z * a4d25a8 2 days ago 
2022-04-14T08:54:09.5243230Z * ee143eb 2 days ago 
2022-04-14T08:54:09.5244048Z | * 5bbf919 2 days ago  (REDACTED)
2022-04-14T08:54:09.5244547Z | * 0f85b30 2 days ago 
2022-04-14T08:54:09.5244774Z | * 844ef3a 2 days ago 
2022-04-14T08:54:09.5245011Z | * abd6960 2 days ago 
2022-04-14T08:54:09.5245206Z |/  
2022-04-14T08:54:09.5245964Z | * 772a918 2 days ago  (REDACTED)
2022-04-14T08:54:09.5246504Z | * c181516 2 days ago 
2022-04-14T08:54:09.5246729Z | * f1f12af 2 days ago 
2022-04-14T08:54:09.5246936Z |/  
2022-04-14T08:54:09.5247602Z | * 4e81a6c 2 days ago  (REDACTED)
2022-04-14T08:54:09.5248066Z | * 81b13d7 2 days ago 
2022-04-14T08:54:09.5248310Z | * d89faa9 2 days ago 
2022-04-14T08:54:09.5248535Z | * e04f090 2 days ago 
2022-04-14T08:54:09.5248775Z | * 3423107 2 days ago 
2022-04-14T08:54:09.5248970Z |/  
2022-04-14T08:54:09.5249605Z | * 103fba4 2 days ago  (REDACTED)
2022-04-14T08:54:09.5250067Z | * ce91a41 2 days ago 
2022-04-14T08:54:09.5250261Z |/  
2022-04-14T08:54:09.5251013Z | * 4af0101 2 days ago  (REDACTED)
2022-04-14T08:54:09.5251528Z | * 2fdeef1 2 days ago 
2022-04-14T08:54:09.5251734Z |/  
2022-04-14T08:54:09.5252460Z | * de822a2 2 days ago  (REDACTED)
2022-04-14T08:54:09.5252959Z | * eb18082 2 days ago 
2022-04-14T08:54:09.5253173Z |/  
2022-04-14T08:54:09.5253372Z * 6586871 3 days ago 
2022-04-14T08:54:09.5253610Z * b9c6931 3 days ago 
2022-04-14T08:54:09.5253832Z * b8c353e 3 days ago 
2022-04-14T08:54:09.5254056Z * 6432d78 3 days ago 
2022-04-14T08:54:09.5254400Z * cf79a59 3 days ago 
2022-04-14T08:54:09.5254624Z * a2ad971 3 days ago 
2022-04-14T08:54:09.5255322Z | * 57f3a5c 3 days ago  (REDACTED)
2022-04-14T08:54:09.5255770Z | * 55c1de9 5 days ago 
2022-04-14T08:54:09.5255966Z |/  
2022-04-14T08:54:09.5256751Z | * e233c89 5 days ago  (REDACTED)
2022-04-14T08:54:09.5257233Z | * 3f6f3b3 5 days ago 
2022-04-14T08:54:09.5257447Z |/  
2022-04-14T08:54:09.5257641Z * 7650e81 6 days ago 
2022-04-14T08:54:09.5257864Z * 08fe9c1 6 days ago 
2022-04-14T08:54:09.5258102Z * f59e5ff 6 days ago 
2022-04-14T08:54:09.5258387Z * 01b5126 6 days ago 
2022-04-14T08:54:09.5258626Z * 6289552 6 days ago 
2022-04-14T08:54:09.5258856Z * a0e9a90 6 days ago 
2022-04-14T08:54:09.5259094Z * 834e427 7 days ago 
2022-04-14T08:54:09.5259323Z * 51d5f88 7 days ago 
2022-04-14T08:54:09.5259548Z * 2f5f906 7 days ago 
2022-04-14T08:54:09.5259785Z * 38a8abf 7 days ago 
2022-04-14T08:54:09.5260009Z * b5360c2 7 days ago 
2022-04-14T08:54:09.5260245Z * d5392ac 7 days ago 
2022-04-14T08:54:09.5260469Z * f09bdf9 7 days ago 
2022-04-14T08:54:09.5260692Z * 5d111d5 8 days ago 
2022-04-14T08:54:09.5260930Z * e099f73 8 days ago 
2022-04-14T08:54:09.5261154Z * 83bd10e 8 days ago 
2022-04-14T08:54:09.5261391Z * b66ace4 8 days ago 
2022-04-14T08:54:09.5261614Z * 4d1c8e7 8 days ago 
2022-04-14T08:54:09.5261835Z * d7c377d 8 days ago 
2022-04-14T08:54:09.5262073Z * 7c621cf 9 days ago 
2022-04-14T08:54:09.5262297Z * 02121ed 9 days ago 
2022-04-14T08:54:09.5262520Z * 782fc41 9 days ago 
2022-04-14T08:54:09.5262758Z * 58560de 9 days ago 
2022-04-14T08:54:09.5263079Z * 9844882 9 days ago 
2022-04-14T08:54:09.5263318Z * 29832d7 9 days ago 
2022-04-14T08:54:09.5263542Z * 3d8424a 9 days ago 
2022-04-14T08:54:09.5263764Z * bf89484 9 days ago 
2022-04-14T08:54:09.5264009Z * 5d9b941 10 days ago 
2022-04-14T08:54:09.5264237Z * 14bf27e 10 days ago 
2022-04-14T08:54:09.5264481Z * bd205a8 10 days ago 
2022-04-14T08:54:09.5264710Z * 7577f18 13 days ago 
2022-04-14T08:54:09.5264937Z * f7c073e 13 days ago 
2022-04-14T08:54:09.5265178Z * 53d20fe 13 days ago 
2022-04-14T08:54:09.5265406Z * 0dcaf92 13 days ago 
2022-04-14T08:54:09.5265645Z * 75f75b6 13 days ago 
2022-04-14T08:54:09.5265751Z 
2022-04-14T08:54:09.5265986Z   INFO [04/14/22 8:54:09:48] Done writing 
2022-04-14T08:54:09.5292836Z ##[error]SyntaxError: Unexpected end of JSON input
2022-04-14T08:54:09.5306897Z ##[section]Finishing: Generate version

```

--- a/src/GitVersion.Core.Tests/BuildAgents/AzurePipelinesTests.cs
+++ b/src/GitVersion.Core.Tests/BuildAgents/AzurePipelinesTests.cs
@@ -80,43 +80,4 @@ public class AzurePipelinesTests : TestBase
         var logMessage = this.buildServer.GenerateSetVersionMessage(vars);
         logMessage.ShouldBe(logPrefix + expectedBuildNumber);
     }
-
-    [Test]
-    public void GetCurrentBranchShouldHandleBranches()
-    {
-        // Arrange
-        this.environment.SetEnvironmentVariable("BUILD_SOURCEBRANCH", $"refs/heads/{MainBranch}");
-
-        // Act
-        var result = this.buildServer.GetCurrentBranch(false);
-
-        // Assert
-        result.ShouldBe($"refs/heads/{MainBranch}");
-    }
-
-    [Test]
-    public void GetCurrentBranchShouldHandleTags()
-    {
-        // Arrange
-        this.environment.SetEnvironmentVariable("BUILD_SOURCEBRANCH", "refs/tags/1.0.0");
-
-        // Act
-        var result = this.buildServer.GetCurrentBranch(false);
-
-        // Assert
-        result.ShouldBeNull();
-    }
-
-    [Test]
-    public void GetCurrentBranchShouldHandlePullRequests()
-    {
-        // Arrange
-        this.environment.SetEnvironmentVariable("BUILD_SOURCEBRANCH", "refs/pull/1/merge");
-
-        // Act
-        var result = this.buildServer.GetCurrentBranch(false);
-
-        // Assert
-        result.ShouldBeNull();
-    }
 }

--- a/src/GitVersion.Core.Tests/BuildAgents/BuildKiteTests.cs
+++ b/src/GitVersion.Core.Tests/BuildAgents/BuildKiteTests.cs
@@ -72,7 +72,7 @@ public class BuildKiteTests : TestBase
         var result = this.buildServer.GetCurrentBranch(false);
 
         // Assert
-        result.ShouldBeNull();
+        result.ShouldBe("refs/pull/55/head");
     }
 
     [Test]

--- a/src/GitVersion.Core.Tests/BuildAgents/CodeBuildTests.cs
+++ b/src/GitVersion.Core.Tests/BuildAgents/CodeBuildTests.cs
@@ -102,30 +102,4 @@ public sealed class CodeBuildTests : TestBase
         props.ShouldContain("GitVersion_Major=1");
         props.ShouldContain("GitVersion_Minor=2");
     }
-
-    [Test]
-    public void GetCurrentBranchShouldHandleTags()
-    {
-        // Arrange
-        this.environment.SetEnvironmentVariable("CODEBUILD_WEBHOOK_HEAD_REF", "refs/tags/1.0.0");
-
-        // Act
-        var result = this.buildServer.GetCurrentBranch(false);
-
-        // Assert
-        result.ShouldBeNull();
-    }
-
-    [Test]
-    public void GetCurrentBranchShouldHandlePullRequests()
-    {
-        // Arrange
-        this.environment.SetEnvironmentVariable("CODEBUILD_SOURCE_VERSION", "refs/pull/1/merge");
-
-        // Act
-        var result = this.buildServer.GetCurrentBranch(false);
-
-        // Assert
-        result.ShouldBeNull();
-    }
 }

--- a/src/GitVersion.Core.Tests/BuildAgents/GitHubActionsTests.cs
+++ b/src/GitVersion.Core.Tests/BuildAgents/GitHubActionsTests.cs
@@ -83,7 +83,7 @@ public class GitHubActionsTests : TestBase
         var result = this.buildServer.GetCurrentBranch(false);
 
         // Assert
-        result.ShouldBeNull();
+        result.ShouldBe("refs/tags/1.0.0");
     }
 
     [Test]
@@ -96,7 +96,7 @@ public class GitHubActionsTests : TestBase
         var result = this.buildServer.GetCurrentBranch(false);
 
         // Assert
-        result.ShouldBeNull();
+        result.ShouldBe("refs/pull/1/merge");
     }
 
     [Test]

--- a/src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs
+++ b/src/GitVersion.Core.Tests/BuildAgents/SpaceAutomationTests.cs
@@ -1,10 +1,11 @@
+using GitVersion;
 using GitVersion.BuildAgents;
 using GitVersion.Core.Tests.Helpers;
 using Microsoft.Extensions.DependencyInjection;
 using NUnit.Framework;
 using Shouldly;
 
-namespace GitVersion.Core.Tests.BuildAgents;
+namespace GitVersionCore.Tests.BuildAgents;
 
 [TestFixture]
 public class SpaceAutomationTests : TestBase
@@ -70,7 +71,7 @@ public class SpaceAutomationTests : TestBase
         var result = this.buildServer.GetCurrentBranch(false);
 
         // Assert
-        result.ShouldBeNull();
+        result.ShouldBe("refs/tags/1.0.0");
     }
 
     [Test]
@@ -83,7 +84,7 @@ public class SpaceAutomationTests : TestBase
         var result = this.buildServer.GetCurrentBranch(false);
 
         // Assert
-        result.ShouldBeNull();
+        result.ShouldBe("refs/pull/1/merge");
     }
 
     [Test]

--- a/src/GitVersion.Core/BuildAgents/AzurePipelines.cs
+++ b/src/GitVersion.Core/BuildAgents/AzurePipelines.cs
@@ -21,17 +21,7 @@ public class AzurePipelines : BuildAgentBase
         $"##vso[task.setvariable variable=GitVersion.{name};isOutput=true]{value}"
     };
 
-    public override string? GetCurrentBranch(bool usingDynamicRepos)
-    {
-        // https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables
-        // BUILD_SOURCEBRANCH does not contain the branch name if the build was triggered by a tag or pull request.
-        string? branchName = Environment.GetEnvironmentVariable("BUILD_SOURCEBRANCH");
-        if (branchName != null && branchName.StartsWith("refs/heads/"))
-        {
-            return branchName;
-        }
-        return null;
-    }
+    public override string? GetCurrentBranch(bool usingDynamicRepos) => Environment.GetEnvironmentVariable("BUILD_SOURCEBRANCH");
 
     public override bool PreventFetch() => true;
 

--- a/src/GitVersion.Core/BuildAgents/BuildKite.cs
+++ b/src/GitVersion.Core/BuildAgents/BuildKite.cs
@@ -30,9 +30,9 @@ public class BuildKite : BuildAgentBase
         }
         else
         {
-            // To align the behavior with the other BuildAgent implementations
-            // we return here also null.
-            return null;
+            // For pull requests BUILDKITE_BRANCH refers to the head, so adjust the
+            // branch name for pull request versioning to function as expected
+            return string.Format("refs/pull/{0}/head", pullRequest);
         }
     }
 

--- a/src/GitVersion.Core/BuildAgents/CodeBuild.cs
+++ b/src/GitVersion.Core/BuildAgents/CodeBuild.cs
@@ -25,17 +25,9 @@ public sealed class CodeBuild : BuildAgentBase
 
     public override string? GetCurrentBranch(bool usingDynamicRepos)
     {
-        string? branchName = Environment.GetEnvironmentVariable(WebHookEnvironmentVariableName);
-        if (string.IsNullOrEmpty(branchName))
-        {
-            branchName = Environment.GetEnvironmentVariable(SourceVersionEnvironmentVariableName);
-        }
+        var currentBranch = Environment.GetEnvironmentVariable(WebHookEnvironmentVariableName);
 
-        if (branchName != null && branchName.StartsWith("refs/heads/"))
-        {
-            return branchName;
-        }
-        return null;
+        return currentBranch.IsNullOrEmpty() ? Environment.GetEnvironmentVariable(SourceVersionEnvironmentVariableName) : currentBranch;
     }
 
     public override void WriteIntegration(Action<string?> writer, VersionVariables variables, bool updateBuildNumber = true)

--- a/src/GitVersion.Core/BuildAgents/GitHubActions.cs
+++ b/src/GitVersion.Core/BuildAgents/GitHubActions.cs
@@ -50,18 +50,7 @@ public class GitHubActions : BuildAgentBase
         }
     }
 
-    public override string? GetCurrentBranch(bool usingDynamicRepos)
-    {
-        // https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
-        // GITHUB_REF must be used only for "real" branches, not for tags and pull requests.
-        // Bug fix for https://github.com/GitTools/GitVersion/issues/2838
-        string? githubRef = Environment.GetEnvironmentVariable("GITHUB_REF");
-        if (githubRef != null && githubRef.StartsWith("refs/heads/"))
-        {
-            return githubRef;
-        }
-        return null;
-    }
+    public override string? GetCurrentBranch(bool usingDynamicRepos) => Environment.GetEnvironmentVariable("GITHUB_REF");
 
     public override bool PreventFetch() => true;
 }

--- a/src/GitVersion.Core/BuildAgents/GitLabCi.cs
+++ b/src/GitVersion.Core/BuildAgents/GitLabCi.cs
@@ -22,9 +22,7 @@ public class GitLabCi : BuildAgentBase
         $"GitVersion_{name}={value}"
     };
 
-    // According to https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
-    // the CI_COMMIT_BRANCH environment variable must be used.
-    public override string? GetCurrentBranch(bool usingDynamicRepos) => Environment.GetEnvironmentVariable("CI_COMMIT_BRANCH");
+    public override string? GetCurrentBranch(bool usingDynamicRepos) => Environment.GetEnvironmentVariable("CI_COMMIT_REF_NAME");
 
     public override bool PreventFetch() => true;
 

--- a/src/GitVersion.Core/BuildAgents/SpaceAutomation.cs
+++ b/src/GitVersion.Core/BuildAgents/SpaceAutomation.cs
@@ -13,15 +13,7 @@ public class SpaceAutomation : BuildAgentBase
 
     protected override string EnvironmentVariable => EnvironmentVariableName;
 
-    public override string? GetCurrentBranch(bool usingDynamicRepos)
-    {
-        string? branchName = Environment.GetEnvironmentVariable("JB_SPACE_GIT_BRANCH");
-        if (branchName != null && branchName.StartsWith("refs/heads/"))
-        {
-            return branchName;
-        }
-        return null;
-    }
+    public override string? GetCurrentBranch(bool usingDynamicRepos) => Environment.GetEnvironmentVariable("JB_SPACE_GIT_BRANCH");
 
     public override string[] GenerateSetParameterMessage(string name, string value) => Array.Empty<string>();
 

